[WIP] Perception

anasarrak opened the pull request.

anasarrak committed:
Re-add ament_package() at the end
If ament_package() is avoided at the end the other modules won't detect moveit_core

anasarrak committed:
Remove colcon_ignore from perception

anasarrak committed:
Update perception CMakeLists and Package.xml to build for ros2

anasarrak committed:
Update perception CMakelists

anasarrak committed:
Starting with ocuppancy_map_monitor

anasarrak committed:
Update occupancy_map_monitor CmakeLists

anasarrak committed:
Update occupancy_map_monitor messages

anasarrak committed:
Add include_directories

anasarrak committed:
Update moveit_core cmakelists

anasarrak committed:
Update perception cmakelists

anasarrak committed:
update includes, perception cmakelists

anasarrak committed:
update libraries occupancy_map_monitor

anasarrak committed:
Update occupancy_map_monitor

anasarrak committed:
Update occupancy_map_updater

anasarrak committed:
Update occupancy_map_server

anasarrak committed:
Add opengl include_dir

anasarrak committed:
Add opengl_include_dir

LanderU committed:
Update Dockerfile

LanderU committed:
Fix Dockerfile for Linux

LanderU committed:
Update external-repos.repos

Added ros1_bridge

LanderU committed:
Update osx-dependencies.sh

* freeglut
* glew

LanderU committed:
Update external-repos.repos

* image_common

ahcorde commented on moveit_core/CMakeLists.txt:
```diff
@@ -18,7 +18,8 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 find_package(rclcpp REQUIRED)
-find_package(Boost REQUIRED)
+find_package(Boost REQUIRED system filesystem date_time thread iostreams)
+# find_package(Boost REQUIRED)
```
Remove this comment you are looking for boost in the previous line

vmayoral replied:
why are we touching moveit_core here for starters? @anasarrak refer to https://github.com/AcutronicRobotics/moveit2/pull/23#issuecomment-471318953 and please keep things tidy in terms of PRs, submit one per each package, it'll make things easier for us upstream.

ahcorde commented on moveit_ros/perception/occupancy_map_monitor/include/moveit/occupancy_map_monitor/occupancy_map_monitor.h:
```diff
@@ -152,12 +155,9 @@ class OccupancyMapMonitor
-
   bool active_;
+
+  // rclcpp::Publisher<octomap_msgs::msg::Octomap> octree_binary_pub_;
```
Why is this commented out?

anasarrak replied:
I was not using it, it was a still WIP, fixed it on https://github.com/AcutronicRobotics/moveit2/commit/92c28f9a441ae4a849d83af31223d35cfc4efdd8

ahcorde commented on moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp:
```diff
@@ -56,127 +57,132 @@ OccupancyMapMonitor::OccupancyMapMonitor(const std::shared_ptr<tf2_ros::Buffer>&
   /* advertise a service for loading octomaps from disk */
-  save_map_srv_ = nh_.advertiseService("save_map", &OccupancyMapMonitor::saveMapCallback, this);
-  load_map_srv_ = nh_.advertiseService("load_map", &OccupancyMapMonitor::loadMapCallback, this);
+  // save_map_srv_ = nh_.advertiseService("save_map", &OccupancyMapMonitor::saveMapCallback, this);
```
Why are you not porting the service? Why are you not porting the service? Do you need some help with this?

anasarrak replied:
Changes made on https://github.com/AcutronicRobotics/moveit2/commit/b04f8b30ea4153be08ce5b53abfb05296dc7442f#diff-ff2eb2808beed252d2672fd10de1b05e

anasarrak replied:
https://github.com/AcutronicRobotics/moveit2/pull/26/files/65ca84acd7b298680785bade43ed81a45657916a#r264123531

anasarrak replied:
https://github.com/AcutronicRobotics/moveit2/commit/b04f8b30ea4153be08ce5b53abfb05296dc7442f

ahcorde commented on moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp:
```diff
@@ -280,7 +286,12 @@ bool OccupancyMapMonitor::getShapeTransformCache(std::size_t index, const std::s
-          ROS_ERROR_THROTTLE(1, "Incorrect mapping of mesh handles");
+          // ROS_ERROR_THROTTLE(1, "Incorrect mapping of mesh handles");
+          //print every sec, something similar to ROS_ERROR_THROTTLE, there is nothing similar on ROS2
+          while (true) {
```
Careful! You cannot used a infinity loop here! You are blocking the program here

anasarrak replied:
Yes, I've changed it to  ``RCUTILS_LOG_ERROR_THROTTLE`` https://github.com/AcutronicRobotics/moveit2/commit/e7f3c1748d8ef863a5e2d5325048ffdd8f84d0c6

ahcorde commented on moveit_ros/perception/occupancy_map_monitor/src/occupancy_map_monitor.cpp:
```diff
@@ -280,7 +286,12 @@ bool OccupancyMapMonitor::getShapeTransformCache(std::size_t index, const std::s
         std::map<ShapeHandle, ShapeHandle>::const_iterator jt = mesh_handles_[index].find(it->first);
         if (jt == mesh_handles_[index].end())
         {
-          ROS_ERROR_THROTTLE(1, "Incorrect mapping of mesh handles");
```
You can use RCUTILS_LOG_ERROR_THROTTLE https://github.com/ros2/rcutils/blob/master/test/test_logging_macros.cpp

anasarrak committed:
Switch error throttle to RCUTILS_LOG_ERROR_THROTTLE

anasarrak committed:
Remove unnecesary Boost components on the CMakeLists

anasarrak committed:
Add moveit_msgs as target_link_libraries

anasarrak committed:
Remove Eigen from package.xml

anasarrak committed:
Add save_map & load_map services

anasarrak committed:
Add auto to node

anasarrak committed:
Wrong node name, fixed

LanderU committed:
Update external-repos.repos

Fix typo

anasarrak committed:
Add shared node

anasarrak committed:
Merge branch 'perception' of https://github.com/AcutronicRobotics/moveit2 into perception

LanderU committed:
Update docker-entrypoint.sh

Added COLCON_IGNORE

LanderU committed:
Fix CI Linux

LanderU committed:
Update osx-compile.sh

try to fix CI OS X

LanderU committed:
Update osx-dependencies.sh:

* libomp

--- stderr: moveit_ros_perception
CMake Error at /usr/local/Cellar/cmake/3.12.3/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)

LanderU committed:
Merge branch 'perception' of https://github.com/AcutronicRobotics/moveit2 into perception

LanderU committed:
Added dependency:

* opencv

CMake Warning at CMakeLists.txt:50 (find_package):
  By not providing "FindOpenCV.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "OpenCV", but
  CMake did not find one.
  Could not find a package configuration file provided by "OpenCV" with any
  of the following names:
    OpenCVConfig.cmake
    opencv-config.cmake

vmayoral committed:
Merge branch 'master' into perception

vmayoral commented:
@anasarrak, can you please address @ahcorde's comments?
